### PR TITLE
Transpile ISA ansatz and hamiltonian

### PR DESCRIPTION
**Pull request by hoaiocom**
Address issue #5 
- Transpile ansatz and op in `get_results_processor` to ISA follow IBMQ update (https://docs.quantum.ibm.com/announcements/news/2024-05-06-ISA-circuit-requirement)
- Use `ansatz_isa` for `get_accuracy_processor`

### CODE REVIEW

**Comment by FelixXu35** [comment]
I would recommend you to include the modification of the variables "dec_state" and "bin_state".

**Comment by hoaiocom** [comment]
Hi @FelixXu35, 
I corrected the order of `bin_state` based on a new layout of all qubits after the ISA transpilation and used that `bin_state` to update the `dec_state` accordingly. 

For example,
`bin_state` =  11101 - `dec_state`= 29 with order [0, 1, 2, 3, 4]
After transpilation (with optimization level = 0): get new order using `final_index_layout` as [2, 4, 0, 1, 3], 
so `new_bin_state` = 11110, and `new_dec_state` = 30.

Please help me check if it is correct in the `get_accuracy_processor` and 2 new supported functions in `utility`
Thank you.

**Review by FelixXu35 — CHANGES_REQUESTED** [review]

File: hamiltoniq/utility.py [hunk 1]
```diff
@@ -57,3 +58,31 @@ def simple_coupling_map() -> list[Tuple[int, int]]:
...
+def get_transpiled_index_layout(
+    tqc: QuantumCircuit, filter_ancillas: None = True
+) -> List[int]:
+    """Return the transpiled index layout of a circuit.
+    Args:
+        tqc: The circuit to get the transpiled index layout of.
+    Returns:
+        The transpiled index layout of the circuit.
+    """
+    return tqc.layout.final_index_layout(filter_ancillas=filter_ancillas)
```
[the remark this answers is not in the document]
Comment (hoaiocom) [reply hamiltoniq/utility.py]: I removed it

File: hamiltoniq/utility.py [hunk 2]
```diff
@@ -57,3 +58,31 @@ def simple_coupling_map() -> list[Tuple[int, int]]:
...
+    Reorder the bits of a binary string.
+    Args:
+        binary: The binary string to reorder.
+        new_order: The new order of the bits.
+    Returns:
+        The binary string with the bits reordered.
+    """
+    binary_str = str(binary)
+    # Create a new binary string based on the new order
+    new_binary = "".join(binary_str[i] for i in new_order)
```
[the remark this answers is not in the document]
Comment (hoaiocom) [reply hamiltoniq/utility.py]: I reverted the get_accuracy_processor and utility to their originals and kept the addition of pm in the get_results_processor. Now doing the test with FakeBackend.

[the remark this answers is not in the document]
Comment (hoaiocom) [reply hamiltoniq/utility.py]: Fixed 2 small bugs:
1. TypeError: Toniq.get_accuracy_processor() takes 4 positional arguments but 5 were given
-> Remove last param [f"qubits_{n_qubits}"] in `accuracy_list = self.get_accuracy_processor(results_list, n_qubits, n_layers)`
2. AttributeError: 'QAOAAnsatz' object has no attribute 'bind_parameters'. Did you mean: '_parameters'?
-> Replaced with `assign_parameters`

Now the `processor_run` can work with FakeBackend as normal.

For example:
```
fake_backend = FakePerth()
score = toniq.processor_run(
            backend=fake_backend,
            n_qubits=3,
            n_layers=2,
            n_reps=10,
        )
```
yield the result of the score=0.8301213540139474, running in 150.179521s

**Pull request merged**